app/waf/src/controllers/schema.ts

```diff
@@ -1,181 +1,67 @@
+import {Entity} from '@loopback/repository';
 import {ParameterObject, ParameterLocation} from '@loopback/openapi-v3-types';
 import {MetadataInspector} from '@loopback/metadata';
 import {PropertyDefinition, PropertyMap} from '@loopback/repository';
-import {Adc} from '../models';
 
-function buildProperties(
-  metadata: PropertyMap | undefined,
-  propNames: string[],
-): {[key: string]: object} {
-  let props: {[key: string]: object} = {};
-
-  if (metadata) {
-    for (let propName of propNames) {
-      props[propName] = {
-        type: metadata[propName].type,
-      };
-    }
-  }
-
-  return props;
+class SchemaProperties {
```

Comment:
There's no need to keep this `SchemaProperties`. 

Very simple way: Find respective parts from `propertyMetadata`, return object directly. 

Your code is time cost, for each Schema.xxxx(), it has to calculate all `create` `update`, `response`... .

Reply:
Done

Reply:
I change the latest implementation back to the original idea, that is to parse property metadta once and save it.